MinPlatformPkg/Test/Library/TestPointCheckLib/DxeCheckLoadedImage.c

```diff
@@ -18,157 +19,14 @@ SPDX-License-Identifier: BSD-2-Clause-Patent
-      ASSERT (FALSE);
-    }
-    if (Subsystem == EFI_IMAGE_SUBSYSTEM_EFI_RUNTIME_DRIVER) {
-      return TRUE;
-    }
-  }
-
-  return FALSE;
-}
-
-VOID
-DumpLoadedImage (
-  IN UINTN                                  Index,
-  IN EFI_LOADED_IMAGE_PROTOCOL              *LoadedImage,
-  IN EFI_DEVICE_PATH_PROTOCOL               *DevicePath,
-  IN EFI_DEVICE_PATH_PROTOCOL               *LoadedImageDevicePath
-  )
-{
-  CHAR16                            *Str;
-  CHAR8                             *PdbPointer;
-
-  DEBUG ((DEBUG_INFO, "[0x%04x]:", Index));
-  if (IsRuntimeImage (LoadedImage->ImageBase)) {
-    DEBUG ((DEBUG_INFO, " (RT)"));
-  } else {
-    DEBUG ((DEBUG_INFO, "     "));
-  }
-  DEBUG ((DEBUG_INFO, " 0x%016lx-0x%016lx", (UINT64)(UINTN)LoadedImage->ImageBase, LoadedImage->ImageSize));
-
-  if (LoadedImageDevicePath != NULL) {
-    Str = ConvertDevicePathToText(LoadedImageDevicePath, TRUE, TRUE);
-    DEBUG ((DEBUG_INFO, " LoadedImageDevicePath=%s", Str));
-    if (Str != NULL) {
-      FreePool (Str);
-    }
-  } else {
-    if (LoadedImage->FilePath != NULL) {
-      Str = ConvertDevicePathToText(LoadedImage->FilePath, TRUE, TRUE);
-      DEBUG ((DEBUG_INFO, " FilePath=%s", Str));
-      if (Str != NULL) {
-        FreePool (Str);
-      }
-    }
-
-    if (DevicePath != NULL) {
-      Str = ConvertDevicePathToText(DevicePath, TRUE, TRUE);
-      DEBUG ((DEBUG_INFO, " DevicePath=%s", Str));
-      if (Str != NULL) {
-        FreePool (Str);
-      }
-    }
-  }
-
-  PdbPointer = PeCoffLoaderGetPdbPointer (LoadedImage->ImageBase);
-  if (PdbPointer != NULL) {
-    DEBUG ((DEBUG_INFO, " (pdb - %a)", PdbPointer));
-  }
-
-  DEBUG ((DEBUG_INFO, "\n"));
-}
+#include "TestPointImageDump.h"
 
 EFI_STATUS
 TestPointCheckNon3rdPartyImage (
-  IN EFI_LOADED_IMAGE_PROTOCOL              *LoadedImage,
-  IN EFI_DEVICE_PATH_PROTOCOL               *DevicePath,
-  IN EFI_DEVICE_PATH_PROTOCOL               *LoadedImageDevicePath
-  )
-{
-  if (LoadedImageDevicePath != NULL) {
-    // LoadedImageDevicePath should be Fv()/FvFile()
-    if (DevicePathType (LoadedImageDevicePath) == MEDIA_DEVICE_PATH &&
-        DevicePathSubType (LoadedImageDevicePath) == MEDIA_PIWG_FW_VOL_DP) {
-      return EFI_SUCCESS;
-    }
-  } else {
-    if (LoadedImage->FilePath != NULL) {
-      // LoadedImage->FilePath should be FvFile()
-      if (DevicePathType (LoadedImage->FilePath) == MEDIA_DEVICE_PATH &&
-          DevicePathSubType (LoadedImage->FilePath) == MEDIA_PIWG_FW_FILE_DP) {
-        return EFI_SUCCESS;
-      }
-    }
-    if (DevicePath != NULL) {
-      // DevicePath should be Fv()
-      if (DevicePathType (DevicePath) == MEDIA_DEVICE_PATH &&
-          DevicePathSubType (DevicePath) == MEDIA_PIWG_FW_VOL_DP) {
-        return EFI_SUCCESS;
-      }
-    }
-  }
-  return EFI_INVALID_PARAMETER;
-}
+  IN EFI_LOADED_IMAGE_PROTOCOL  *LoadedImage,
+  IN EFI_DEVICE_PATH_PROTOCOL   *DevicePath,
+  IN EFI_DEVICE_PATH_PROTOCOL   *LoadedImageDevicePath
+  );
```

Comment:
It would be nice to have some of these function prototypes defined in header files so each potential C file doesn't need to do define it (which creates N copies of the prototype to keep in sync).

I don't think it is required for this review, it was already a problem. Please just keep in mind in case you're aware of places you feel need this now.

Reply:
I agree.  I moved this to the corresponding header file.